docs/nsm_api.md

```diff
@@ -215,13 +274,19 @@ message VniRange {
  * remote_mechanism_type - remote mechanism seleted by NSM2 from the options presented by
  *                         NSM1
  * remote_mechanism_parameters - parameters selected by NSM2 for the remote mechanism
+ * labels - key value pairs to communicate arbitrary context about the reply
  */
 message RemoteConnectionReply {
-    required string request_id = 1;
-    required bool accepted = 2;
-    optional string admission_error = 3;
-    required RemoteMechanismType remote_mechanism_type = 4;
-    required RemoteMechanismParameters remote_mechanism_parameters= 5;
+    string request_id = 1;
+    bool accepted = 2;
+    RemoteMechanismType remote_mechanism_type = 4;
+    RemoteMechanismParameters remote_mechanism_parameters= 5;
+    ConnectionContext connection_context = 6;
+
+    /* fields below are optional */
+
+    map<string,string> labels = 7;
+    string admission_error = 3;
```

Comment:
please change position to line 282.

Reply:
admission error cannot be optional.

Reply:
@sbezverk What if there is no admission error because you succeeded?  Perhaps it needs to go above the 'optional' things but below the 'required' things with a comment indicating its required for failure but otherwise optional?

Reply:
@edwarnicke then it is nil